research-software-directory/RSD-as-a-service
# New setup for saving images

# Refactor image storage and reference

**This issue is work in progress.**

Closes #631 #635 #636 #656

Changes proposed in this pull request:
* Upgrade to Next v13
* Images are stored in one table. The id is SHA-1 of the image data. This addresses the caching in the most optimal way for an SPA (reloading of page is not required when user changes image).
* Search for contributors and team members extended with avatar image. Note! This feature is only supported for RSD persons as ORCID api does not provide avatar images.
* Search for contributors and team members in RSD includes both contributors and team members
* Data generation script is updated to support new image approach
 
How to test:
- `make start` to rebuild and generate some random data
- validate image uploads on all places:
- for software: contributors avatar upload, organisation logo upload
- for projects: project image, team member avatar upload, organisation logo upload
- for organisations: organisation logo upload, research unit organisation logo uploadPR Checklist
- add a person as software contributor, then go to project to team members and look for this person. You should be able to find this person in RSD
- when person with avatar is removed from all references the avatar image should also be removed from image table
- when project image is removed from all references the image should also be removed from image table
